pkg/trait/knative_test.go

```diff
@@ -361,6 +362,204 @@ func TestKnativeTriggerConfiguration(t *testing.T) {
 		matching = matching && assert.Equal(t, "/events/evt.type", trigger.Spec.Subscriber.URI.Path)
 		matching = matching && assert.Equal(t, "default-test-evttype", trigger.Name)
 
+		matching = matching && assert.NotNil(t, trigger.Spec.Filter)
+		matching = matching && assert.Len(t, trigger.Spec.Filter.Attributes, 2)
+		matching = matching && assert.Equal(t, trigger.Spec.Filter.Attributes["type"], "evt.type")
+		matching = matching && assert.Equal(t, trigger.Spec.Filter.Attributes["source"], "my-source")
+
+		return matching
+	}))
+}
+
+func TestKnativeTriggerConfigurationDefaultEventTypeFilter(t *testing.T) {
+	catalog, err := camel.DefaultCatalog()
+	require.NoError(t, err)
+
+	c, err := NewFakeClient("ns")
+	require.NoError(t, err)
+
+	traitCatalog := NewCatalog(c)
+
+	environment := Environment{
+		CamelCatalog: catalog,
+		Catalog:      traitCatalog,
+		Client:       c,
+		Integration: &v1.Integration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "ns",
+			},
+			Status: v1.IntegrationStatus{
+				Phase: v1.IntegrationPhaseDeploying,
+			},
+			Spec: v1.IntegrationSpec{
+				Profile: v1.TraitProfileKnative,
+				Sources: []v1.SourceSpec{
+					{
+						DataSpec: v1.DataSpec{
+							Name: "route.java",
+							Content: `
+								public class CartoonMessagesMover extends RouteBuilder {
+									public void configure() {
+										from("knative:event/evt.type")
+ 											.log("${body}");
+									}
+								}
+							`,
+						},
+						Language: v1.LanguageJavaSource,
+					},
+				},
+				Traits: v1.Traits{
+					Knative: &traitv1.KnativeTrait{
+						Trait: traitv1.Trait{
+							Enabled: pointer.Bool(true),
+						},
+					},
+				},
+			},
+		},
+		IntegrationKit: &v1.IntegrationKit{
+			Status: v1.IntegrationKitStatus{
+				Phase: v1.IntegrationKitPhaseReady,
+			},
+		},
+		Platform: &v1.IntegrationPlatform{
+			Spec: v1.IntegrationPlatformSpec{
+				Cluster: v1.IntegrationPlatformClusterOpenShift,
+				Build: v1.IntegrationPlatformBuildSpec{
+					PublishStrategy: v1.IntegrationPlatformBuildPublishStrategyS2I,
+					Registry:        v1.RegistrySpec{Address: "registry"},
+					RuntimeVersion:  catalog.Runtime.Version,
+				},
+				Profile: v1.TraitProfileKnative,
+			},
+			Status: v1.IntegrationPlatformStatus{
+				Phase: v1.IntegrationPlatformPhaseReady,
+			},
+		},
+		EnvVars:        make([]corev1.EnvVar, 0),
+		ExecutedTraits: make([]Trait, 0),
+		Resources:      k8sutils.NewCollection(),
+	}
+	environment.Platform.ResyncStatusFullConfig()
+
+	// don't care about conditions in this unit test
+	_, err = traitCatalog.apply(&environment)
+
+	require.NoError(t, err)
+	assert.NotEmpty(t, environment.ExecutedTraits)
+	assert.NotNil(t, environment.GetTrait("knative"))
+
+	assert.NotNil(t, environment.Resources.GetKnativeTrigger(func(trigger *eventing.Trigger) bool {
```

Comment:
Why not taking the explicit trigger and doing each assert specifically. If that fails we know exactly what's going on.